### PR TITLE
Add make fmt-check as a way to check if go fmt is needed

### PR DESCRIPTION
**Pull request by bn222**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -113,6 +113,13 @@ generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and
 fmt: ## Run go fmt against code.
 	go fmt ./...
 
+.PHONY: fmt-check
+fmt-check:
+	@files=$$(find . -name "*.go" -not -path "./vendor/*" -not -path "./dpu-api/vendor/*"); \
+	output=$$(gofmt -d $$files); \
```
Comment (wizhaoredhat) [inline Makefile:119]: Typo? Shouldn't there be a space between go and fmt

Comment (bn222) [reply Makefile:119]: no, gofmt is capable to do a dry run. `go fmt` isn't.

Comment (wizhaoredhat) [reply Makefile:119]: Ah I see. you want to be able to dry run to see what go fmt will do to the code.

**Pull request merged**